src/storage/QueryBoundProcessor.cpp

```diff
@@ -13,94 +13,204 @@
 namespace nebula {
 namespace storage {
 
-kvstore::ResultCode QueryBoundProcessor::processVertex(PartitionID partId,
-                                                       VertexID vId) {
-    FilterContext fcontext;
+kvstore::ResultCode QueryBoundProcessor::processEdgeImpl(const PartitionID partId,
+                                                         const VertexID vId,
+                                                         const EdgeType edgeType,
+                                                         const std::vector<PropContext>& props,
+                                                         FilterContext& fcontext,
+                                                         cpp2::VertexData& vdata) {
+    RowSetWriter rsWriter;
+    auto ret = collectEdgeProps(
+        partId, vId, edgeType, props, &fcontext,
+        [&, this](RowReader* reader, folly::StringPiece k, const std::vector<PropContext>& p) {
+            RowWriter writer(rsWriter.schema());
+            PropsCollector collector(&writer);
+            this->collectProps(reader, k, p, &fcontext, &collector);
+            rsWriter.addRow(writer);
+        });
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        return ret;
+    }
+
+    if (!rsWriter.data().empty()) {
+        vdata.edge_data.emplace_back(apache::thrift::FragileConstructor::FRAGILE, edgeType,
+                                     std::move(rsWriter.data()));
+    }
+
+    return ret;
+}
+
+kvstore::ResultCode QueryBoundProcessor::processAllEdge(PartitionID partId, VertexID vId,
+                                                        FilterContext& fcontext,
+                                                        cpp2::VertexData& vdata) {
+    auto prefix = NebulaKeyUtils::prefix(partId, vId);
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto ret = kvstore_->prefix(spaceId_, partId, prefix, &iter);
+    if (ret != kvstore::ResultCode::SUCCEEDED || !iter) {
+        return ret;
+    }
+
+    std::unordered_set<EdgeType> edgeTypes;
+    for (; iter->valid(); iter->next()) {
+        auto key = iter->key();
+        if (NebulaKeyUtils::isEdge(key)) {
+            auto edgeType = NebulaKeyUtils::getEdgeType(key);
+            if (edgeType < 0) {
+                continue;
+            }
+            std::vector<PropContext> props;
+
+            auto ite = edgeTypes.find(edgeType);
+            if (ite != edgeTypes.end()) {
+                continue;
+            }
+
+            edgeTypes.emplace(edgeType);
+            auto it = edgeContexts_.find(edgeType);
+            if (it == edgeContexts_.end()) {
+                PropContext pc;
+                cpp2::PropDef pd;
+                pc.pikType_ = PropContext::PropInKeyType::DST;
+                pc.type_.type = nebula::cpp2::SupportedType::INT;
+                pd.name = "_dst";
```

Comment:
Why return '_dst' by default?

Reply:
If so,   why not return '_edgeType',  '_rank'?  Because <src, edgeType, rank, dst> identify one edge.

Reply:
This is just to deal with GO statements, such as "go from 1 over *" , this statement needs to return "_dst" for all edges.

Reply:
I mean for sentence "go from 1 over * ",  we should return the whole edge key,   just '_dst' is not enough

Reply:
If all the default properties are returned, it is not compatible with the return of a single edge.

For example: "go from 1 over edge1", this sentence will return edge1._dst.